insights/parsers/freeradius.py

```diff
@@ -0,0 +1,53 @@
+"""
+FreeRadius - Freeradius configuration files
+===========================================
+
+FreeRadius is a parser for the freeradius configuration files usually in
+``/etc/raddb``.
+Files ending with ``.rpmnew`` are excluded because they are backups made when
+ugprading but customer modified the configuration files, so are not used by the
+service.
+
+Because this parser reads multiple files, the files are stored as a list within
+the parser and need to be iterated through in order to find specific
+configuration files.
+
+Examples:
+
+    >>> for cfgfile in shared[FreeRadius]: # Parser contains list of all configuration files
+    ...     print "Path:", cfgfile.file_path
+    ...
+    Path: etc/raddb/radiusd.conf
+    Path: etc/raddb/proxy.conf
+    Path: etc/raddb/policy.d/filter
+
+"""
+
+from .. import Parser, parser
+from . import SkipException
+
+
+@parser("freeradius")
+class FreeRadius(Parser):
+    """
+    Parse freeradius configuration files. Each item is a dictionary with
+    following fields:
+
+    Attributes:
+        file_name (str): configuration file name
+        file_path (str): configuration file path without leading '/'
+        file_content (list): configuration file content, line by line
+        file_content_stripped (list): stripped configuration file content, line by line
+    """
+
+    def parse_content(self, content):
+        # Ignore ".rpmnew" files, these are backups made when upgrading but
+        # customer modified the configuration files
+        if self.file_name.endswith('.rpmnew'):
+            raise SkipException("Skipping '.rpmnew' file.")
+        self.file_content = content
+        self.file_content_stripped = []
+        for l in content:
+            s = l.strip()
+            if s and not s.startswith('#'):
+                self.file_content_stripped.append(s)
```

Comment:
Personally I would like this to do a lot more.  There are many other good examples of configuration file parsers that store structured information.  Key/value pairs as a dictionary?  Structured information?  Go for it!

And I'm really not sure what the point of storing the whole content _as well as_ only the active lines.  What are you intending to do with it?

Reply:
IOW _DO NOT_ leave the rule writer to implement your parsing for you.